packages/flutter_tools/test/general.shard/artifacts_test.dart

```diff
@@ -545,4 +545,31 @@ void main() {
       );
     });
   });
+
+  group('LocalEngineInfo', () {
+    late FileSystem fileSystem;
+    late LocalEngineInfo localEngineInfo;
+
+    setUp(() {
+      fileSystem = MemoryFileSystem.test();
+    });
+
+    testUsingContext('determines the target device name from the path', () {
```

Comment:
These tests are failing because you're using the zone injected context `globals.fs` but not overriding it. Instead, you could make `LocalEngineInfo` accept the filesystem in the constructor.

Reply:
Thanks! Went with the former approach for now but open to reconsidering.